Filtration/ViewModels/ItemFilterScriptViewModel.cs

```diff
@@ -235,6 +269,22 @@ private void AddItemFilterBlockViewModels(IEnumerable<IItemFilterBlockBase> item
                     SelectedBlockViewModel = vm;
                     firstNewViewModel = false;
                 }
+
+                var itemBlock = itemFilterBlock as IItemFilterBlock;
+                if (itemBlock != null)
+                {
+                    foreach (var blockItem in itemBlock.BlockItems)
+                    {
+                        var customSoundBlockItem = blockItem as CustomSoundBlockItem;
+                        if (customSoundBlockItem != null)
+                        {
```

Comment:
You can replace this with `foreach(var customSoundBlockItem in itemBlock.BlockItems.OfType<CustomSoundBlockItem>)` rather than manually casting each item and checking its type.